docs/reference/indices/simulate-template.asciidoc

```diff
@@ -0,0 +1,308 @@
+
+////
+[source,console]
+--------------------------------------------------
+PUT _component_template/component_template1
+{
+  "template": {
+    "mappings": {
+      "properties": {
+        "@timestamp": {
+          "type": "date"
+        }
+      }
+    }
+  }
+}
+
+PUT _component_template/other_component_template
+{
+  "template": {
+    "mappings": {
+      "properties": {
+        "ip_address": {
+          "type": "ip"
+        }
+      }
+    }
+  }
+}
+
+PUT _index_template/template_1
+{
+  "index_patterns": ["te*", "bar*"],
+  "template": {
+    "settings": {
+      "number_of_shards": 1
+    },
+    "mappings": {
+      "_source": {
+        "enabled": false
+      },
+      "properties": {
+        "host_name": {
+          "type": "keyword"
+        },
+        "created_at": {
+          "type": "date",
+          "format": "EEE MMM dd HH:mm:ss Z yyyy"
+        }
+      }
+    },
+    "aliases": {
+      "mydata": { }
+    }
+  },
+  "priority": 10,
+  "composed_of": ["component_template1", "other_component_template"],
+  "version": 3,
+  "_meta": {
+    "description": "my custom"
+  }
+}
+--------------------------------------------------
+// TESTSETUP
+
+[source,console]
+--------------------------------------------------
+DELETE _index_template/*
+DELETE _component_template/*
+--------------------------------------------------
+// TEARDOWN
+////
+
+[source,console]
+--------------------------------------------------
+POST /_index_template/_simulate/template_1
+--------------------------------------------------
+
+[[simulate-template-api-request]]
+==== {api-request-title}
+
+`POST /_index_template/_simulate/<index-template>`
+
+
+[[simulate-template-api-path-params]]
+==== {api-path-parms-title}
+
+`<index-template>`::
+(Optional, string)
+The name of the index template to simulate.
+To simulate an arbitrary template configuration, 
+omit this parameter and specify the template configuration in the request body.
+
+[[simulate-template-api-query-params]]
+==== {api-query-parms-title}
+
+`cause`::
+(Optional, string) The reason for using the specified template for the simulation. 
+
+`create`::
+(Optional, boolean) If `true`, the template passed in the body is 
+only used if no existing template matches the same index pattern. 
+If `false`, the template with the highest priority is used for the simulation.
+Note that the template is not permanently added or updated in either case, 
+it is only used for the simulation.
+Defaults to `false`.
+
+include::{es-ref-dir}/rest-api/common-parms.asciidoc[tag=master-timeout]
+
+[[simulate-template-api-request-body]]
+==== {api-request-body-title}
+
+`composed_of`::
+(Optional, array of strings)
+An ordered list of component template names. Component templates are merged in the order
+specified, meaning that the last component template specified has the highest precedence. See 
+<<multiple-component-templates,Composing multiple component templates>> for an example.
```

Comment:
Nit: I tend to stick phrases like `for an example` in the front of the sentence.
```suggestion
specified, meaning that the last component template specified has the highest precedence. 
For an example, see 
<<multiple-component-templates,Composing multiple component templates>>.
```